receiver/prometheusreceiver/internal/metricsbuilder.go

```diff
@@ -87,6 +88,23 @@ func (b *metricBuilder) matchStartTimeMetric(metricName string) bool {
 
 // AddDataPoint is for feeding prometheus data complexValue in its processing order
 func (b *metricBuilder) AddDataPoint(ls labels.Labels, t int64, v float64) error {
+	// Any datapoint with duplicate labels MUST be rejected per:
+	// * https://github.com/open-telemetry/wg-prometheus/issues/44
+	// * https://github.com/open-telemetry/opentelemetry-collector/issues/3407
+	// as Prometheus rejects such too as of version 2.16.0, released on 2020-02-13.
+	seen := make(map[string]bool)
+	dupLabels := make([]string, 0, len(ls))
+	for _, label := range ls {
+		if _, ok := seen[label.Name]; ok {
+			dupLabels = append(dupLabels, label.Name)
```

Comment:
Do we need to include the full list of duplicate labels in the error returned or can we simply short circuit here once we know there is at least one duplicate label?

Reply:
I becomes a UX debugging nightmare if you have to clear one duplicate, redeploy your process then onto the next one. It is trivial for us to list all the duplicate label names, and get a much better user experience.